consensus/types/src/blob_sidecar.rs

```diff
@@ -150,10 +150,10 @@ impl<E: EthSpec> BlobSidecar<E> {
         })
     }
 
-    pub fn new_with_existing_proof(
+    pub fn new_with_existing_proof<Payload: AbstractExecPayload<E>>(
```

Comment:
I didn't get how we are using this

Reply:
Previously this method requires full payload, but the new code i added passes a block with blinded payload, so I've made this generic to support both.

https://github.com/sigp/lighthouse/pull/6755/files#diff-8b85bc6756dc32dfe7385f21235257abddf0c3ad5f36ca66a4a4078e1209970bR307-R310